tests/model/write_model_test.go

```diff
@@ -17,6 +17,46 @@ var testCases = map[string]struct {
 	model string
 	code  int
 }{
+	// implemented in Fails_If_Using_Self_As_Type_Name
```

Comment:
Can we uncomment these and use them?

Reply:
I can't because the DSL parsing fails :( 

```
=== RUN   TestWriteAuthorizationModel/case1
    --- FAIL: TestWriteAuthorizationModel/case1 (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x10067fe75]

goroutine 33 [running]:
testing.tRunner.func1.2({0x100c55720, 0x1017eab60})
	/usr/local/opt/go/libexec/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	/usr/local/opt/go/libexec/src/testing/testing.go:1399 +0x39f
panic({0x100c55720, 0x1017eab60})
	/usr/local/opt/go/libexec/src/runtime/panic.go:884 +0x212
github.com/craigpastro/openfga-dsl-parser/v2.(*dslListener).ExitTypeDefinition(0xc0003c0d00, 0x100fb8bb0?)
	/Users/mparnisari/go/pkg/mod/github.com/craigpastro/openfga-dsl-parser/v2@v2.0.0-20230110230133-e2e9547a2b86/parser.go:50 +0x35
github.com/craigpastro/openfga-dsl-parser/v2/internal/gen/dsl/parser.(*TypeDefinitionContext).ExitRule(0xc0003b9d20?, {0x100fb8bb0?, 0xc0003c0d00?})
	/Users/mparnisari/go/pkg/mod/github.com/craigpastro/openfga-dsl-parser/v2@v2.0.0-20230110230133-e2e9547a2b86/internal/gen/dsl/parser/dsl_parser.go:415 +0x45
github.com/antlr/antlr4/runtime/Go/antlr/v4.(*ParseTreeWalker).ExitRule(0xc0001247f8?, {0x100fb8bb0, 0xc0003c0d00}, {0x10a7b5460?, 0xc000170ff0?})
	/Users/mparnisari/go/pkg/mod/github.com/antlr/antlr4/runtime/!go/antlr/v4@v4.0.0-20221202181307-76fa05c21b12/tree.go:249 +0x69
github.com/antlr/antlr4/runtime/Go/antlr/v4.(*ParseTreeWalker).Walk(0xc0001247f8?, {0x100fb8bb0, 0xc0003c0d00}, {0x100fbb7e8?, 0xc000170ff0})
	/Users/mparnisari/go/pkg/mod/github.com/antlr/antlr4/runtime/!go/antlr/v4@v4.0.0-20221202181307-76fa05c21b12/tree.go:233 +0x165
github.com/antlr/antlr4/runtime/Go/antlr/v4.(*ParseTreeWalker).Walk(0xc0001401c0?, {0x100fb8bb0, 0xc0003c0d00}, {0x100fbb680?, 0xc0001247f8})
	/Users/mparnisari/go/pkg/mod/github.com/antlr/antlr4/runtime/!go/antlr/v4@v4.0.0-20221202181307-76fa05c21b12/tree.go:231 +0x108
github.com/craigpastro/openfga-dsl-parser/v2.Parse({0x100e3fdf3, 0x4a})
	/Users/mparnisari/go/pkg/mod/github.com/craigpastro/openfga-dsl-parser/v2@v2.0.0-20230110230133-e2e9547a2b86/parser.go:227 +0x3cb
github.com/craigpastro/openfga-dsl-parser/v2.MustParse(...)
	/Users/mparnisari/go/pkg/mod/github.com/craigpastro/openfga-dsl-parser/v2@v2.0.0-20230110230133-e2e9547a2b86/parser.go:204
github.com/openfga/openfga/tests/model.runTests.func1(0x100fbedf8?)
	/Users/mparnisari/GitHub/openfga/tests/model/write_model_test.go:377 +0x94
testing.tRunner(0xc0005831e0, 0xc000179630)
	/usr/local/opt/go/libexec/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	/usr/local/opt/go/libexec/src/testing/testing.go:1493 +0x35f
```